shared/keys/cmd/create.go

```diff
@@ -0,0 +1,163 @@
+)
+
+// createCmd represents the create command
+var createCmd = &cobra.Command{
+	Use:   "create <name>",
+	Short: "Creating an encrypted private key and save to <name> file as the key pair identifier",
+	Long: `Derive a new private key and encrypt to disk.
+
+Allow users to use BIP39 mnemonic and set passphrases to secure the mnemonic. Supports BIP32 Hierarchical
+Deterministic (HD) path to derive a specific account. Take key ID <name> from user and store key under the <name>.
+Key file is encrypted with the given password (required).
+	
+	--dry-run	Generate/Recover a key without stored it to the local keystore.
+
+	-i			Prompt the user for BIP44 path, BIP39 mnemonic, and passphrase.
+
+	--recover 	Recover a key from a seed passphrase.
+`,
+	Args: cobra.ExactArgs(1),
+	RunE: runAddCmdPrepare,
+}
+
+func runAddCmdPrepare(cmd *cobra.Command, args []string) error {
+	clientCtx, err := client.GetClientQueryContext(cmd)
+	if err != nil {
+		return err
+	}
+
+	buf := bufio.NewReader(clientCtx.Input)
+	return runAddCmd(clientCtx, cmd, args, buf)
+}
+
+/*
+input
+  - bip39 mnemonic
+  - bip39 passphrase
+  - bip44 path
+  - local encryption password
+
+output
+  - armor encrypted private key (saved to file)
+*/
+func runAddCmd(ctx client.Context, cmd *cobra.Command, args []string, inBuf *bufio.Reader) error {
+	var err error
+
+	// [Miniature Keybase] Using level.db for keybase and ED25519 for public/private key generation
+
+	name := args[0]
+
+	// TODO: determine proper keystore location later
+	kb, err := leveldb.OpenFile("./.keybase/poktKeys.db", nil)
+
+	// Creating key with ED25519
+	pubKey, privKey, err := ed25519.GenerateKey(nil) // TODO: using mnemonic to generate (read)
+	keyMap := make(map[string][]byte)
+	keyMap["pubKey"] = pubKey
+	keyMap["privKey"] = privKey
+
+	// TODO: bette print function (support Test and JSON)
+	fmt.Printf("Name (KEY UID): %s\n", name)
+	fmt.Printf("Public key: %v\n", pubKey)
+	fmt.Printf("Private key: %v\n", privKey)
+
+	// Store the raw key to keybase (TODO: check if key already exists; user input sanitize)
+
+	// TODO: store JSON formated name -> {pubkey, privkey, ...}
+	err = kb.Put([]byte(name), privKey, nil)
+	if err != nil {
+		return err
+	}
+
+	// TODO: mnemonic key generation and recovery feature
+
+	// Get bip39 mnemonic
+	var mnemonic string
+	var bip39Passphrase string = ""
+
+	recover, _ := cmd.Flags().GetBool(flagRecover)
+	if recover {
+		mnemonic, err = input.GetString("Enter your bip39 mnemonic", inBuf)
+		if err != nil {
+			return err
+		}
+
+		if !bip39.IsMnemonicValid(mnemonic) {
+			return errors.New("invalid mnemonic")
+		}
+	} else {
+		// read entropy seed straight from tmcrypto.Rand and convert to mnemonic
+		entropySeed, err := bip39.NewEntropy(mnemonicEntropySize)
+		if err != nil {
+			return err
+		}
+
+		mnemonic, err = bip39.NewMnemonic(entropySeed)
+		if err != nil {
+			return err
+		}
+	}
+
+	// TODO: Using Mnemonic to generate keys
+
+	// TODO: Import the CryptoPocket!
```

Comment:
`TODO: (<Issue-#>): Import CryptoPocket`

Please add follow up issue numbers - also, why can't this be done `IN_THIS_COMMIT`?

Reply:
I just have the crypto functions updated, and this is done in this commit now.